src/lib.rs

```diff
@@ -70,8 +70,13 @@
 
 #![deny(warnings, missing_docs, missing_debug_implementations)]
 #![doc(html_root_url = "https://docs.rs/bytes/0.4")]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I think one of the easiest things to do to ease development of no_std libraries is to unconditionally use the `#![no_std]` attribute.

Only if the `std` feature is enabled should this have `extern crate std`, and then in that case the prelude can be imported to specific modules if necessary.